Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1108,16 +1108,18 @@ class BrowserViewController: UIViewController {
 
     /// Updates the URL bar security, text and button states.
     fileprivate func updateURLBar() {
-        guard let tab = tabManager.selectedTab else { return }
-        if let url = tab.url, url.isLocal {
+        guard let tab = tabManager.selectedTab, let url = tab.url else { return }
+        if url.isLocal {
             self.topToolbar.locationView.rewardsButton.isVerified = false
+        } else {
+            self.rewards?.ledger.fetchPublisherActivity(from: url, faviconURL: nil, publisherBlob: nil, tabId: UInt64(tab.rewardsId))
         }
         
-        topToolbar.currentURL = tab.url?.displayURL
+        topToolbar.currentURL = url.displayURL
```

Comment:
This logic change worries me a bit. It used to be if the URL was nil for whatever reason, we'd be emptying the current url and do other stuff below, but now you guard the entire method based on existence of that URL. Can we maybe isolate this change-set to not alter other logic in this function?